### PR TITLE
Add API for confirmation mail

### LINKED ISSUE #34: Write an API to send confirmation email
Reference: https://docs.djangoproject.com/en/3.0/topics/email/

### PR DESCRIPTION
**Pull request by SomTambe**
Closes #34  .

- [x] Added generate_verification_code() function in User model.

- [x] Added ActivationMailer API function.

- [x] Removed ***/migrations/* from .gitignore.

- [x] Removed CircularDependency Error.  

- [x] Add 'verify/' link in users.urls .

- [x] Added settings_email.py for all email settings.

- [x] Added CSRF_COOKIE_SECURE = True argument. Change this to False when testing API and putting requests from your machine.

While testing API, remember to uncomment @csrf_exempt in users.models .

### CODE REVIEW

File: users/views.py [hunk 1]
```diff
@@ -43,3 +50,18 @@ def post(self, request):
...
+        recipient = user_data.email
+        name = user_data.name
+        user_code = user_data.generate_verification_code()
+        user_link = ACTIVATION_LINK.format(code = user_code)
+        subject = ACTIVATION_SUBJECT
+        body = ACTIVATION_BODY.format(name=name, link=user_link)
+        send_mail(subject, body, sender, [recipient], fail_silently=False)
+        return redirect(ACTIVATION_REDIRECT)
+    else :
+        return HttpResponse("Invalid request!", status=400)
```
Comment (vikrantmalik051) [inline users/views.py]: Add newline here.

File: users/urls.py [hunk 2]
```diff
@@ -1,7 +1,10 @@
...
 from django.urls import path
+from django.views.decorators.csrf import csrf_exempt
 
 urlpatterns = [
     path('auth/login/', LoginView.as_view(), name="auth-login"),
     path('auth/logout/', LogoutView.as_view(), name="auth-logout"),
-]
+    path('verify/', ActivationMailer, name="activation-verify")
+    # path('verify/', csrf_exempt(ActivationMailer))
+]
```
Comment (vikrantmalik051) [inline users/urls.py]: Add newline here.

File: campusdiscussbackend/settings_email.py [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+EMAIL_PORT = 5432
+EMAIL_HOST_USER = 'example@email.com'
+EMAIL_HOST_PASSWORD = 'password'
+
+ACTIVATION_SUBJECT = "Activation mail for Campus Discuss"
+ACTIVATION_BODY = """Hi {name:s}! 
+                     Click on the followiing link or cpoy-paste it to continue with the activation procedure.
+                     {link:s}."""
+ACTIVATION_REDIRECT = "https://redirectlink"
+ACTIVATION_LINK = "https://campusdiscuss.in/verify?code={code:s}"
```
Comment (vikrantmalik051) [inline campusdiscussbackend/settings_email.py]: Add newline here.

**Comment by SomTambe** [comment]
Added newline.

**Comment by SomTambe** [comment]
Added users.tests for API testing. 
Changed a minor spelling-mistake in settings_email.py.
Added a try-except block in ActivationMailer() API.

**Pull request merged**